### ISSUE TITLE
Reusable `ILabStatus` in downstream distributions

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/jupyterlab#10415]
### Problem

Currently the `status` defined on the `JupyterLab` application here:

https://github.com/jupyterlab/jupyterlab/blob/96b6ca109fb194d1f011a2f6fe1f4c673d88e6ea/packages/application/src/lab.ts#L102-L105

Refers to a `LabStatus` defined here: 

https://github.com/jupyterlab/jupyterlab/blob/96b6ca109fb194d1f011a2f6fe1f4c673d88e6ea/packages/application/src/status.ts#L57-L60

The `ILabStatus` token is provided by this plugin:

https://github.com/jupyterlab/jupyterlab/blob/96b6ca109fb194d1f011a2f6fe1f4c673d88e6ea/packages/application-extension/src/index.tsx#L862-L875

However there isn't too many JupyterLab specific requirements in `ILabStatus` / `LabStatus`. Most of the logic actually operates on a `JupyterFrontEnd`:

https://github.com/jupyterlab/jupyterlab/blob/96b6ca109fb194d1f011a2f6fe1f4c673d88e6ea/packages/application/src/status.ts#L57-L141

### Proposed Solution

To avoid confusion around the name, we could rename `ILabStatus` to `IStatus` or `IApplicationStatus`.

It's also possible that we don't need the explicit check for a `JupyterLab` in some of the plugins like the following:

https://github.com/jupyterlab/jupyterlab/blob/96b6ca109fb194d1f011a2f6fe1f4c673d88e6ea/packages/application-extension/src/index.tsx#L868-L870

### Additional context

This could make it easier to reuse the `ILabStatus` in alternate JupyterLab distributions such as RetroLab: https://github.com/jupyterlab/retrolab/pull/166

**Comment by jtpio** [comment]
Adding to 4.0 for consideration although it's more of a nice-to-have, and also because of the API changes that change would introduce.

**Comment by jtpio** [comment]
I looked into this issue a little bit and while `ILabStatus` could indeed be renamed to `IApplicationStatus`, the check for the `JupyterLab` instance would still be needed because `status` is not part of the `JupyterFrontEnd.IShell` interface:

https://github.com/jupyterlab/jupyterlab/blob/96b6ca109fb194d1f011a2f6fe1f4c673d88e6ea/packages/application-extension/src/index.tsx#L868-L870

Which means it would not be possible to reuse that plugin as is in downstream applications like Notebook 7: https://github.com/jupyter/notebook. So doing the rename to `IApplicationStatus` might not be worth it.

Also the amount of code for exposing a custom `ILabStatus` in a downstream plugin is not that big. Linking to https://github.com/jupyterlab/retrolab/pull/166 as an example where it was formerly added to RetroLab.

Closing this issue for now, although happy to reconsider it later if someone wants to discuss it more.
